### PR TITLE
Build on windows using rwinlib and BH

### PR DESCRIPTION
**Pull request by jeroen**
Not saying that this is how you need to do it, but at least this will work on win builder so you can download the binary packages put them on your drat.

### CODE REVIEW

File: src/Makevars.win [hunk 1]
```diff
@@ -1,17 +1,12 @@
-# -*- mode: Makefile -*-
-#
-# Copyright 2005 - 2006 Dominick Samperi
-# Copyright 2005        Uwe Ligges
-# Copyright 2008 - 2015 Dirk Eddelbuettel
-# Copyright 2011        Uwe Ligges, Brian Ripley, and Josh Ulrich
-
```
Comment (eddelbuettel) [inline src/Makevars.win]: It's a little unusual to remove old attribution so I'll probably put this back in.

File: DESCRIPTION [hunk 2]
```diff
@@ -13,10 +13,9 @@ Depends: R (>= 2.10.0)
 Suggests: rgl, RUnit, shiny
 LazyLoad: true
 Imports: methods, Rcpp (>= 0.11.0), stats, graphics, zoo
-LinkingTo: Rcpp
+LinkingTo: BH, Rcpp
```
Comment (eddelbuettel) [inline DESCRIPTION]: I have to think this through.  I am not sure yet if that is good.  Maybe using the same Boost that QL was build with is better.

But this helps for win-builder.  Maybe I just fold it in and out as needed.

**Comment by eddelbuettel** [comment]
And then it passes.

So with that can you please make three changes to the PR:
1. Pretty-please add [the first seven lines of the current src/Makevars.win](https://github.com/eddelbuettel/rquantlib/blob/master/src/Makevars.win#L1-L7)
2. Remove BH from LinkingTo
3. Maybe add OS_type: unix back in (I am ambivalent about this)

All this gives me a way to build a binary and may keep RQL still off CRAN.

**Comment by jeroen** [comment]
Without BH we don't have the required boost headers. Also R will refuse to build anything on Windows that has `OS_type: unix`.

**Comment by eddelbuettel** [comment]
With BH you bomb on Travis.  Travis has had [Boost -dev packages installed](https://github.com/eddelbuettel/rquantlib/blob/master/.travis.yml#L16) for years, else it would not work.

What I am basically suggesting that the PR to `master` restores the Linux functionality. I will then (at least for now) "manually" mod DESCRIPTION to play on Win-Builder.  

Does that make sense?

**Comment by eddelbuettel** [comment]
And you rock. I was just about to ask you to add the Boost headers to the quantlib library in rwinlib, and I see you are already on it.  :1st_place_medal:

**Comment by jeroen** [comment]
I doubt this will fix it but we'll see. Boost is 120MB of headers alone 😨

**Comment by jeroen** [comment]
Yes if we can somehow extract the subset of boost headers that quantlib needs that would relieve some of the weight.

**Comment by eddelbuettel** [comment]
Glance at [this script used by BH](https://github.com/eddelbuettel/bh/blob/master/local/scripts/CreateBoost.sh#L96-L138).  Maybe if we just point at 'test'?

**Comment by jeroen** [comment]
Is it specified anywhere which `boostlibs` are used by quantlib?

**Comment by eddelbuettel** [comment]
'Base' (not sure what the name is) for boost::shared_ptr and all that, the boost test framework and maybe one more. 

For the Debian package I add the same two as I add for Travis: `libboost-dev, libboost-test-dev`

**Comment by eddelbuettel** [comment]
It does have a lot, as there are interdependencies.  But there are a dozen more `libboost$FOO-dev`.  See:

```
edd@rob:~$ apt show libboost-all-dev
Package: libboost-all-dev
Version: 1.62.0.1
Priority: optional
Section: universe/libdevel
Source: boost-defaults
Origin: Ubuntu
Maintainer: Ubuntu Developers <ubuntu-devel-discuss@lists.ubuntu.com>
Original-Maintainer: Debian Boost Team <pkg-boost-devel@lists.alioth.debian.org>
Bugs: https://bugs.launchpad.net/ubuntu/+filebug
Installed-Size: 10.2 kB
Depends: libboost-dev, libboost-tools-dev, libboost-atomic-dev, libboost-chrono-dev, libboost-context-dev, libboost-coroutine-dev, libboost-date-time-dev, libboost-exception-dev, libboost-fiber-dev, libboost-filesystem-dev, libboost-graph-dev, libboost-graph-parallel-dev, libboost-iostreams-dev, libboost-locale-dev, libboost-log-dev, libboost-math-dev, libboost-mpi-dev, libboost-mpi-python-dev, libboost-program-options-dev, libboost-python-dev, libboost-random-dev, libboost-regex-dev, libboost-serialization-dev, libboost-signals-dev, libboost-system-dev, libboost-test-dev, libboost-thread-dev, libboost-timer-dev, libboost-type-erasure-dev, libboost-wave-dev
Homepage: http://www.boost.org/
Download-Size: 2,230 B
APT-Manual-Installed: yes
APT-Sources: http://us.archive.ubuntu.com/ubuntu artful/universe amd64 Packages
Description: Boost C++ Libraries development files (ALL) (default version)
 The Boost web site provides free, peer-reviewed, portable C++ source
 libraries. The emphasis is on libraries which work well with the C++
 Standard Library. One goal is to establish "existing practice" and
 provide reference implementations so that the Boost libraries are
 suitable for eventual standardization. Some of the libraries have
 already been proposed for inclusion in the C++ Standards Committee's
 upcoming C++ Standard Library Technical Report.
 .
 This metapackage provides the complete Boost development environment,
 including all separately-packaged libraries.
 .
 This package is a dependency package, which depends on Debian's default
 Boost version (currently 1.62).

edd@rob:~$ 
```

**Pull request merged**